managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -3811,9 +3811,11 @@ public int applyMaxSizeCap(int maxEntries, long maxSizeBytes) {
         if (maxSizeBytes == NO_MAX_SIZE_LIMIT) {
             return maxEntries;
         }
-        int maxEntriesBasedOnSize =
-                Long.valueOf(estimateEntryCountBySize(maxSizeBytes, readPosition, ledger)).intValue();
-        return Math.min(maxEntriesBasedOnSize, maxEntries);
+        long estimatedEntryCount = estimateEntryCountBySize(maxSizeBytes, readPosition, ledger);
+        if (estimatedEntryCount > Integer.MAX_VALUE) {
```

Comment:
I have a question that could `estimateEntryCountBySize` return a negative value? If so, should we check `estimatedEntryCount < 0` as well?

Reply:
This is a value estimated from `maxSizeBytes`, if it's invalid (either negative or a value bigger than a 32-bits integer), we should treat it as unlimited.

Reply:
The estimateEntryCountBySize method returns this `return Math.max(result, 1);`, the value shouldn't be less than 1.